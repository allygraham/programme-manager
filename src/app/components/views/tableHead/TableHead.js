import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

import { constants } from '../../../constants';

const ClickableTableHead = styled('th')`
  cursor: pointer;
  min-width: 5rem;
  transition: color 300ms ease-in-out;

  &:hover {
    color: #3273DC;
  }
`;

const SortIcon = styled('i')`
  margin-left: 0.5rem;
`;

export const TableHead = ({ sortBy, sortCriteria }) => (
  <tr>
    <ClickableTableHead
      onClick={() => sortBy('id')}
      active={sortCriteria === 'id'}
    >
      {constants.tableTitles.id}
      {sortCriteria === 'id' && <SortIcon className="fa fa-chevron-down"/>}
    </ClickableTableHead>
    <ClickableTableHead
      onClick={() => sortBy('name')}
      active={sortCriteria === 'name'}
    >
      {constants.tableTitles.name}
      {sortCriteria === 'name' && <SortIcon className="fa fa-chevron-down"/>}
    </ClickableTableHead>
    <th>{constants.tableTitles.description}</th>
    <th>{constants.tableTitles.status}</th>
    <th>{constants.tableTitles.remove}</th>
  </tr>
);

TableHead.propTypes = {
  sortBy: PropTypes.func.isRequired,
  sortCriteria: PropTypes.string.isRequired,
};
