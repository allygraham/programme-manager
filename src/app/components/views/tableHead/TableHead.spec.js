import React from 'react';
import { shallow } from 'enzyme';

import { TableHead } from './TableHead';

describe('TableHead', () => {

  test('render the <TableHead> component', () => {
    const sortBy = jest.fn();
    const sortCriteria = 'name';

    const wrapper = shallow(<TableHead sortBy={sortBy} sortCriteria={sortCriteria} />);
    expect(wrapper).toMatchSnapshot();
  });

});