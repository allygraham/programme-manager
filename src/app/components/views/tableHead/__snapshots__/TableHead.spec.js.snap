// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHead render the <TableHead> component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TableHead
    sortBy={[MockFunction]}
    sortCriteria="name"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Styled(th)
          active={false}
          onClick={[Function]}
        >
          ID
        </Styled(th)>,
        <Styled(th)
          active={true}
          onClick={[Function]}
        >
          Name
          <Styled(i)
            className="fa fa-chevron-down"
          />
        </Styled(th)>,
        <th>
          Description
        </th>,
        <th>
          Status
        </th>,
        <th>
          Remove
        </th>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "children": Array [
            "ID",
            false,
          ],
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "ID",
          false,
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": true,
          "children": Array [
            "Name",
            <Styled(i)
              className="fa fa-chevron-down"
            />,
          ],
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Name",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "fa fa-chevron-down",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Description",
        },
        "ref": null,
        "rendered": "Description",
        "type": "th",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Status",
        },
        "ref": null,
        "rendered": "Status",
        "type": "th",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Remove",
        },
        "ref": null,
        "rendered": "Remove",
        "type": "th",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Styled(th)
            active={false}
            onClick={[Function]}
          >
            ID
          </Styled(th)>,
          <Styled(th)
            active={true}
            onClick={[Function]}
          >
            Name
            <Styled(i)
              className="fa fa-chevron-down"
            />
          </Styled(th)>,
          <th>
            Description
          </th>,
          <th>
            Status
          </th>,
          <th>
            Remove
          </th>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "children": Array [
              "ID",
              false,
            ],
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "ID",
            false,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": true,
            "children": Array [
              "Name",
              <Styled(i)
                className="fa fa-chevron-down"
              />,
            ],
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Name",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "fa fa-chevron-down",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Description",
          },
          "ref": null,
          "rendered": "Description",
          "type": "th",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Status",
          },
          "ref": null,
          "rendered": "Status",
          "type": "th",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Remove",
          },
          "ref": null,
          "rendered": "Remove",
          "type": "th",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
